ci/functional/test_main_prep_node.sh

```diff
@@ -2,33 +2,226 @@
+        return 0
+      fi
+      sleep ${retry_wait}
+    done
+}
+
+# First check for infinband devices
+for i in $(seq 0 $ib_count); do
     iface="ib$i"
-    if [ -e /sys/class/net/"$iface" ]; then
-        if ! ifconfig "$iface" | grep "inet "; then
-          {
-            echo "Found interface $iface down after reboot on $HOSTNAME"
-            systemctl status || true
-            systemctl --failed || true
-            journalctl -n 500 || true
-            ifconfig "$iface" || true
-            cat /sys/class/net/"$iface"/mode || true
-            ifup "$iface" || true
-            ifconfig -a || true
-            ls -l /etc/sysconfig/network-scripts/ifcfg-* || true
-            cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
-          } 2>&1 | mail -s "Interface $iface found down after reboot" \
-                   -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL"
+    if do_wait_for_ib "$iface"; then
+        set +x
+        if ! ip addr show "$iface" | grep "inet "; then
+            ib_message="$({
+                echo "Found interface $iface down after reboot on $HOSTNAME"
+                ip addr show "$iface" || true
+                cat /sys/class/net/"$iface"/mode || true
+                ip link set up "$iface" || true
+                cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
+                } 2>&1)"
+            mail_message+="${nl}${ib_message}${nl}"
+            echo "$ib_message"
+        else
+            echo "OK: Interface $iface is up."
         fi
-        if ! ifconfig "$iface" | grep "inet "; then
+        if ! ip addr show "$iface" | grep "inet "; then
+            echo "Fail: $ib_message"
+            result=1
             echo "Failed to bring up interface $iface on $HOSTNAME. " \
-            "Please file a CORCI ticket."
-            exit 1
+            "Please file a SRE ticket."
         fi
+        if [ -e "/sys/class/net/$iface/device/numa_node" ]; then
+            set -x
+            cat "/sys/class/net/$iface/device/numa_node"
+        fi
+        set -x
+    else
+        ib_message="Failed to bring up interface $iface on $HOSTNAME. "
+        mail_message+="${nl}${ib_message}${nl}"
+        echo "$ib_message"
+        result=1
+        echo "Please file a SRE ticket."
     fi
 done
 
-if ! grep /mnt/share /proc/mounts; then
+# having -x just makes the console log harder to read.
+set +x
+if [ -e /sys/class/net/ib1 ]; then
+    # now check for pmem & NVMe drives when ib1 is present.
+    # ipmctl show -dimm should show an even number of drives, all healthy
+    dimm_count=0
+    while IFS= read -r line; do
+        if [[ "$line" != *"| Healthy "* ]]; then continue; fi
+        ((dimm_count=dimm_count+1))
+    done < <(ipmctl show -dimm)
+    if [ $dimm_count -eq 0 ] || [ $((dimm_count%2)) -ne 0 ]; then
+       dimm_message="FAIL: Wrong number $dimm_count healthy PMEM DIMMs seen."
+       mail_message+="$nl$dimm_message$nl$(ipmctl show -dimm)$nl"
```

Comment:
Should there be a `result=false` here?

Reply:
No.  It is not a fatal condition to the running test.  It just means that the DIMM can no longer self repair when the next "sector" needs to be replaced.   So this is just gets an email so we can schedule the downtime.